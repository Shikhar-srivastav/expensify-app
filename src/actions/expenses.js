import { v4 as uuidv4 } from 'uuid';

export const addExpense = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        _id: uuidv4(),
        description,
        note,
        amount,
        createdAt
    }
});

export const removeExpense = (_id) => ({
    type: 'REMOVE_EXPENSE',
    _id
});

export const editExpense = (_id, updates) => ({
    type: 'EDIT_EXPENSE',
    _id,
    updates
});