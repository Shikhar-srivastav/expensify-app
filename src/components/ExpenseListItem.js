import React from 'react';
import { Link, resolvePath } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = props => (
    <div>
        <Link to={`/edit/${props._id}`}>
            <h3>{props.description}</h3>
        </Link>
        <p>
            {`Rs${numeral(props.amount).format('0,0.00')} - ${moment(props.createdAt).format('Do MMMM YYYY')}`}
        </p>
    </div>
);

export default ExpenseListItem;