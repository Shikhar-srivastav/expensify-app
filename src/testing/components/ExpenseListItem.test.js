import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Should render ExpenseList Item', () => {
    const expense = expenses[0];
    const wrapper = shallow(
        <ExpenseListItem
            key={expense._id}
            {...expense}
        />
    );
    expect(wrapper).toMatchSnapshot();
});