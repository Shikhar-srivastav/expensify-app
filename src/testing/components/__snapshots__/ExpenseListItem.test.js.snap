// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList Item 1`] = `
<div>
  <Link
    to="./edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    10
     - 
    0
  </p>
</div>
`;
